### DISCUSSION TITLE
Livewire Flickity Conflict/Issue

### DISCUSSION THREAD

**Discussion by mansoorkhan96** [livewire/livewire#5065] [1. Help] [answered]
Hello,

I am having issues rendering a dynamic slider built using [Flickity](https://flickity.metafizzy.co/). I have a CRUD where I add/remove slider images to the database and load them instantly live on the page. I have the following code setup, when I add a new slider image using `addPhoto()` method, it is added but does not load on the page until I refresh the page.

1) Livewire and Flickity conflict with each other. If i remove `wire:ignore` from the blade template it loads the newly added photo but messes with the Flickity slider (of course livewire manipulates the DOM).

2) If i keep `wire:ignore` inside the blade template it does not load the newly added photo until i manually reload the page. I also tried the '$refresh' via listener but no luck.

Is there any solution or workaround for this issue? 

```php
class ProfileEdit extends Component
{
    public $photos;

    public function mount()
    {
        $this->loadPhotos();
    }

    public addPhoto()
    {
        // Add a new photo to database

        // Load latest photos into the property 
        $this->loadPhotos();
    }

    protected function loadPhotos()
    {
        $this->photos = load photos from database
    }

    public function render()
    {
        return view('livewire.carousel');
    }
}
```

```blade
<div>
    <div wire:ignore x-data="carousel()" x-init="init()" x-ref="carousel">
        @foreach ($photos as $photo)
            <div wire:key="{{ $loop->index }}">
                
            </div>
        @endforeach
    </div>

    <script>
       function carousel() {
            return {
                init() {
                    var flkty = new Flickity(this.$refs.carousel, {
                        // 
                    });
                },
            }
        } 
    </script>
</div>
```

**Comment by joshhanley** [comment, marked as answer]
@mansoorkhan96 yeah that happens a lot with javascript libraries and Livewire, due to the library modifying the DOM and Livewire not knowing about it and wiping out the changes, hence why you need `wire:ignore`.

That does mean though that looping through the images won't be updated as Livewire is ignoring everything inside that div.

Couple of possible work arounds:

You could try using `wire:ignore.self`, that may work, it depends on how much DOM flickity modifies.
Another option would be to make use of Alpine and `@entangle` to keep the photos array in sync in javascript and use that to loop over them.

Hope this helps!

**Comment by codevApp** [comment]
@mansoorkhan96 I have the same problem and have been trying to find a solution for hours. Have you been able to solve it?

**Reply by mansoorkhan96** [reply]
Hi,
`wire:ignore` and `@entangle` worked for me.

**Reply by codevApp** [reply]
Hi, 
thanks for your help.

tried it:

        <div wire:ignore x-data="carousel()" x-init="init()" x-ref="carousel">
            <template x-for="image in images">
                <div class="carousel-cell">
                    <img class="carousel-cell-img" :src="image.url"  alt="" src=""/>
                </div>
            </template>
        </div>


        function carousel() {
            return {
                images: @entangle('images').live,
                init() {
                    var flkty = new Flickity(this.$refs.carousel, {
                            prevNextButtons: true,
                            pageDots: false,
                            fullscreen: true,
                    });
                },
            }
        }

It works and the src change. But the flicktiy slider does not work. I think because the first tag is the <template x-for ..> and not a div with the img tag in it.

Can you help? How did you solve this?

**Reply by mansoorkhan96** [reply]
@codevApp I had the project abandoned and no longer have it setup.

And just realised that, It never worked for me and I force reloaded the page after each upload.

You can try to [watch](https://alpinejs.dev/magics/watch) the `images` object using Alpine and then everytime the images gets changed, just re-initialize/re-mount the carousel.

**Reply by mansoorkhan96** [reply]
Try this:

```html
<div
    wire:ignore
    x-data="{
        images: @entangle('images').live,
        initFlickity() {
            var flkty = new Flickity(this.$refs.carousel, {
                    prevNextButtons: true,
                    pageDots: false,
                    fullscreen: true,
            });
        },
    }"
    x-ref="carousel"
    x-init="initFlickity()"
    x-init="$watch('images', value => initFlickity())"
>
    <template x-for="image in images">
    </template>
</div>
```

**Reply by codevApp** [reply]
Hi, this works and the images are also updated. Only because of the <template x-for tag Flickity does not work. I have made two screenschots so you can see what I mean.

This is with alpine and template x-for. Slider not working but Image Url´s changing
![Screenshot 2024-02-26 195940 111](https://github.com/livewire/livewire/assets/42936773/06125911-780b-40e2-8786-0c7c0d342806)

This is with Blade foreach, slider working. I think Flickity will change the style on the carousel-cell
![Flicktiy2](https://github.com/livewire/livewire/assets/42936773/72cd98e3-420a-445a-ae1b-a2ed2e852765)

The only woraround so far would be to "reload" the page via wire:navigate redirect :/ But that is not the elegant solution.

**Reply by codevApp** [reply]
I gave up and took splidejs. It works right away. Thanks again anyway!

https://splidejs.com/
https://alpinejs.dev/component/splide
